freqtrade/misc.py

```diff
@@ -228,3 +230,24 @@ def parse_db_uri_for_logging(uri: str):
         return uri
     pwd = parsed_db_uri.netloc.split(':')[1].split('@')[0]
     return parsed_db_uri.geturl().replace(f':{pwd}@', ':*****@')
+
+
+def get_strategy_run_id(strategy: 'IStrategy') -> str:
+    """
+    Generate unique identification hash for a backtest run. Identical config and strategy file will
+    always return an identical hash.
+    :param strategy: strategy object.
+    :return: hex string id.
+    """
+    digest = hashlib.sha1()
+    digest.update(rapidjson.dumps(strategy.config, default=str,
+                                  number_mode=rapidjson.NM_NATIVE).encode('utf-8'))
+    with open(strategy.__file__, 'rb') as fp:
+        digest.update(fp.read())
+    return digest.hexdigest().lower()
+
+
+def get_backtest_metadata_filename(filename: Union[Path, str]) -> Path:
+    """Return metadata filename for specified backtest results file."""
+    base, ext = os.path.splitext(filename)
```

Comment:
I would rather stick to pathlib - which should have this available as `filename.suffix` (obviously assuming the input is Path ...)

os.path can have ... questionable behaviour on certain platforms (windows, i'm looking at you) - so i don't trust it for anything related to paths anymore